Extensions/PathfindingBehavior/tests/legacypathfindingruntimebehavior.spec.js

```diff
@@ -0,0 +1,689 @@
+        targetY = playerY - 2 * cellSize;
+        obstacleX = playerX;
+        obstacleY =
+          targetY - playerTopBorder - obstacleBottomBorder - inclusiveMargin;
+        break;
+      case 'down':
+        targetX = playerX;
+        targetY = playerY + 2 * cellSize;
+        obstacleX = playerX;
+        obstacleY =
+          targetY + playerBottomBorder + obstacleTopBorder + inclusiveMargin;
+        break;
+    }
+
+    obstacle.setPosition(obstacleX, obstacleY);
+    // so obstacles register
+    runtimeScene.renderAndStep(1000 / 60);
+
+    // move as close as possible to the obstacle
+    player.setPosition(playerX, playerY);
+    player.getBehavior(pathFindingName).moveTo(runtimeScene, targetX, targetY);
+    expect(player.getBehavior(pathFindingName).pathFound()).to.be(true);
+    expect(player.getBehavior(pathFindingName).getNodeCount()).to.be(3);
+  });
+
+  it("mustn't find a path with an obstacle slightly overlaping to the target", function () {
+    let obstacleX;
+    let obstacleY;
+    let targetX;
+    let targetY;
+    switch (direction) {
+      case 'right':
+        targetX = playerX + 2 * cellSize;
+        targetY = playerY;
+        obstacleX =
+          targetX +
+          playerRightBorder +
+          obstacleLeftBorder +
+          inclusiveMargin -
+          1;
+        obstacleY = playerY;
+        break;
+      case 'left':
+        targetX = playerX - 2 * cellSize;
+        targetY = playerY;
+        obstacleX =
+          targetX -
+          playerLeftBorder -
+          obstacleRightBorder -
+          inclusiveMargin +
+          1;
+        obstacleY = playerY;
+        break;
+      case 'up':
+        targetX = playerX;
+        targetY = playerY - 2 * cellSize;
+        obstacleX = playerX;
+        obstacleY =
+          targetY -
+          playerTopBorder -
+          obstacleBottomBorder -
+          inclusiveMargin +
+          1;
+        break;
+      case 'down':
+        targetX = playerX;
+        targetY = playerY + 2 * cellSize;
+        obstacleX = playerX;
+        obstacleY =
+          targetY +
+          playerBottomBorder +
+          obstacleTopBorder +
+          inclusiveMargin -
+          1;
+        break;
+    }
+
+    obstacle.setPosition(obstacleX, obstacleY);
+    // so obstacles register
+    runtimeScene.renderAndStep(1000 / 60);
+
+    // move as close as possible to the obstacle
+    player.setPosition(playerX, playerY);
+    player.getBehavior(pathFindingName).moveTo(runtimeScene, targetX, targetY);
+    expect(player.getBehavior(pathFindingName).pathFound()).to.be(false);
+  });
+
+  const obstacleTop = obstacle;
+  const obstacleBottom = addObstacleSprite3(
+    runtimeScene,
+    objectCenteredOnCells
+  );
+
+  it('can find a path between 2 ajusted obstacles', function () {
+    let topObstacleX;
+    let bottomObstacleX;
+    let topObstacleY;
+    let bottomObstacleY;
+    let targetX;
+    let targetY;
+    let straitLineCells;
```

Comment:
typo: straightLineCells

Also, make it obvious this is a variable containing a count, not containing "cells" or a complicated thing:

`straightLineCellCount`